data-products/src/main/scala/org/sunbird/analytics/job/report/CourseMetricsJob.scala

```diff
@@ -73,26 +78,19 @@ object CourseMetricsJob extends optional.Application with IJob with ReportGenera
     spark.read.format("org.apache.spark.sql.cassandra").options(settings).load()
   }
 
-  def getActiveBatches(loadData: (SparkSession, Map[String, String]) => DataFrame, druidQuery: DruidQueryModel)(implicit spark: SparkSession, fc: FrameworkContext): Array[Row] = {
+  def getActiveBatches(loadData: (SparkSession, Map[String, String]) => DataFrame, config: JobConfig)(implicit spark: SparkSession, fc: FrameworkContext): Array[Row] = {
     implicit val sqlContext = new SQLContext(spark.sparkContext)
     import sqlContext.implicits._
 
-    val courseBatchDF = loadData(spark, Map("table" -> "course_batch", "keyspace" -> sunbirdCoursesKeyspace))
-      .select("courseid", "batchid", "enddate", "startdate")
     val timestamp = DateTime.now().minusDays(1).withTimeAtStartOfDay()
 
-    /**
-      * 1. Fetching course_id and channel from content-model-snapshot
-      * 2. Mapping it with course_batch details
-      */
-
-    val druidResult = recordTime(DruidDataFetcher.getDruidData(druidQuery), "Total time taken to fetch from druid: ")
-    val finalResult = druidResult.map { f => JSONUtils.deserialize[DruidOutput](f) }
-    val finalDF = recordTime(finalResult.toDF(),"Time taken to convert to DF: ")
+    val batchInfo = getTPDBatches(config)
+    val courseBatchDenormDF = batchInfo.toDF().withColumn("batchInfo", explode(col("batches")))
+      .withColumn("batchId",col("batchInfo").getItem("batchId"))
```

Comment:
Better to maintain a unique name in the job. since I see few places we are referring "batchId" as camelCase and few place we are referring "batchid" as smaller case. it will confuse other developers.
Before doing any join operation and once we read the data from DB convert all columns names into the unique format. 
Also define all these columns names as constant and use that object variable in both assessment and course job
`val columns = {
courseId = "courseid/courseId" // any one
}`